### ISSUE TITLE
Add all Request States as filters on Advanced Search

### ISSUE THREAD

**Issue by lmullane** [bcgov/foi-flow#4921]
* As an IAO analyst or ministry coordinator
* I want to filter Advanced Search by any request state
* so that I can more quickly find requests

**Assumptions & Scope**	
What are the assumptions for this story?

Requested by CFD who have trouble easily finding request on their request queue but applicable to all users.

What is IN scope?
Add all request states as clickable search boxes on Advanced Search. Currently on these states: Unopened, Call for Records, Records Review, Ministry Sign Off, Closed

Add these states: Intake in Progress, Redirect, Section 5, Application Fees, Open, Fee Estimate, On Hold, Deduplication, Harms Assessment, Peer Review, Consult, Response

What is NOT in scope?
Clicking a checkbox is already completed

**Acceptance Criteria**	

**Scenario 1: All Request States**
* GIVEN an IAO analyst or ministry coordinator is logged in
* WHEN they are on Advanced Search
* THEN they will see all request states with a checkbox: Unopened, Call for Records, Records Review, Ministry Sign Off, Closed, Intake in Progress, Redirect, Section 5, Application Fees, Open, Fee Estimate, On Hold, Deduplication, Harms Assessment, Peer Review, Consult, Response

**Scenario 2: Apply search **
* GIVEN an IAO analyst or ministry coordinator is logged in
* AND they have clicked a box next to a request state: Unopened, Call for Records, Records Review, Ministry Sign Off, Closed, Intake in Progress, Redirect, Section 5, Application Fees, Open, Fee Estimate, On Hold, Deduplication, Harms Assessment, Peer Review, Consult, Response
* WHEN they click the Apply Search button
* THEN only the requests in the selected state will appear in the Advanced Search results

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by lmullane** [comment]
@JHarrietha-AOT we will need a design to organize all the request states.

cc @m-prodan

**Comment by m-prodan** [comment]
I wonder if given the volume, a dropdown would make more sense now? I am thinking the space will get too busy if we have too many states as selectable check boxes.

**Comment by JHarrietha-AOT** [comment]
@m-prodan I would agree dropdown with check boxes would be better, but do we think some request states (ie. the ones already in Advanced Search) Are more likely to be searched then the rest? Like should the be a most frequently used and the other with a dropdown or move them all into a dropdown?
Let me know your thoughts and I can get a design together quickly for this.

**Comment by m-prodan** [comment]
I think they can all be in a drop down instead. For most folks, they can technically search by state in their team requests if they wanted a quick picture of their team, and I don't imagine many managers are regularly checking other team's requests by state - at least that's just my thinking on it.

**Comment by JHarrietha-AOT** [comment]
That makes sense to me - I'll get this design done before end of day so we can get this ticket refined

**Comment by lmullane** [comment]
@m-prodan, as discussed give the updated AC a read and let me know if I missed anything?

cc @JHarrietha-AOT

**Comment by JHarrietha-AOT** [comment]
@m-prodan @lmullane as discussed here is the updated design changing the request state selection to a dropdown like all of the others in Advanced Search with the checkboxes for multi selecting. Reach out if you have any questions. 

<img width="1415" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBd1dlQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c58cffe25785187b1d31cddb79c9db2d0924e61f/Dropdown.jpg" alt="Dropdown.jpg" />

**Comment by m-prodan** [comment]
Looks good @lmullane - and I like the corresponding design too @JHarrietha-AOT

**Comment by KyEggleston** [comment]
Dropdown should be checkbox.

**Comment by sumathi-thirumani-aot** [comment]
Internal Technical Refinement (4921):
=====
Estimate: 8

Open Questions:
1. Information required on the option "All Active"
2. UX suggestion for state multiselect - https://mui.com/material-ui/react-select/
**Update from Bus. Refinement:**
1. **There is no option as "All Active". Ignore it**
2. **Follow the multi select component "Public Body".**
